### PR TITLE
Motion planner config

### PR DESCRIPTION
**Pull request by marrts**
Includes changes to allow for more configurability in the motion planner and ability to change configuration without recompiling.

### CODE REVIEW

**Comment by jrgnicho** [comment]
Still WIP?

**Comment by marrts** [comment]
Let me do some cleaning and review this afternoon and then I'll take off the WIP tag

**Comment by marrts** [comment]
No longer WIP. Also note that some of these changes were pulled in from #74, mainly the crs_process_data package files.

**Comment by jrgnicho** [comment]
@marrts can you squash some of these commits?

File: crs_motion_planning/src/motion_planning_server.cpp [hunk 1]
```diff
@@ -264,28 +180,33 @@ class MotionPlanningServer : public rclcpp::Node
   void planProcess(std::shared_ptr<crs_msgs::srv::PlanProcessMotions::Request> request,
                    std::shared_ptr<crs_msgs::srv::PlanProcessMotions::Response> response)
   {
-    crs_motion_planning::pathPlanningConfig motion_planner_config = *motion_planner_config_;
+    namespace fs = boost::filesystem;
+    crs_motion_planning::pathPlanningConfig::Ptr motion_planner_config;
+    std::string config_rel_path = "config/motion_planning/MP_config.yaml";
```
Comment (jrgnicho) [inline crs_motion_planning/src/motion_planning_server.cpp:185]: Can this file path not be passed as a parameter?

File: crs_motion_planning/src/motion_planning_server.cpp [hunk 2]
```diff
@@ -399,30 +321,38 @@ class MotionPlanningServer : public rclcpp::Node
...
   }
 
   void planFreespace(std::shared_ptr<crs_msgs::srv::CallFreespaceMotion::Request> request,
                      std::shared_ptr<crs_msgs::srv::CallFreespaceMotion::Response> response)
   {
     using namespace crs_motion_planning;
-    crs_motion_planning::pathPlanningConfig motion_planner_config = *motion_planner_config_;
+    namespace fs = boost::filesystem;
+    crs_motion_planning::pathPlanningConfig::Ptr motion_planner_config;
+    std::string config_rel_path = "config/motion_planning/MP_config.yaml";
```
Comment (jrgnicho) [inline crs_motion_planning/src/motion_planning_server.cpp:333]: Same as above

File: crs_motion_planning/src/path_planning_utils.cpp [hunk 3]
```diff
@@ -363,6 +562,7 @@ bool crsMotionPlanner::generateSurfacePlans(pathPlanningResults::Ptr& results)
                                     post_speed_split_trajs[i].points.end());
       if (!findClosestJointOrientation(end_orig, new_raster_end, end_new, config_->descartes_config.axial_step))
       {
+        ret_app_addable = false;
```
Comment (jrgnicho) [inline crs_motion_planning/src/path_planning_utils.cpp:565]: This variable name makes it hard to tell how it's used

File: crs_process_data/data/main/toolpaths/part_2/configs/job_0_degrees_20200520T192241.997282_config.yaml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  process_execution:
+    time_tolerance: 5.0
+    joint_tolerance: [0.035, 0.035, 0.035, 0.035, 0.035, 0.035]
+  part_registration:
+    simulation_pose: [-0.7, 0.0, 1.1, 0.0, 0.0, 1.57]
+    seed_pose: [-0.68, 0.0, 1.0, 0.0, 0.0, 1.57]
+    target_frame_id: world
+    part_file: /home/jnicho/crs_data/part_2/part_2.stl
+    toolpath_file: /home/jnicho/crs_data/part_2/job_0_degrees_20200520T192241.997282.yaml
+  part_rework: {}
```
Comment (jrgnicho) [inline crs_process_data/data/main/toolpaths/part_2/configs/job_0_degrees_20200520T192241.997282_config.yaml:37]: did you intent to save this custom config file?

File: crs_process_data/data/surrogate1/toolpaths/part_2/configs/job_0_degrees_20200520T192241.997282_config.yaml [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  process_execution:
+    time_tolerance: 5.0
+    joint_tolerance: [0.035, 0.035, 0.035, 0.035, 0.035, 0.035]
+  part_registration:
+    simulation_pose: [-0.7, 0.0, 1.1, 0.0, 0.0, 1.57]
+    seed_pose: [-0.68, 0.0, 1.0, 0.0, 0.0, 1.57]
+    target_frame_id: world
+    part_file: /home/jnicho/crs_data/part_2/part_2.stl
+    toolpath_file: /home/jnicho/crs_data/part_2/job_0_degrees_20200520T192241.997282.yaml
+  part_rework: {}
```
Comment (jrgnicho) [inline crs_process_data/data/surrogate1/toolpaths/part_2/configs/job_0_degrees_20200520T192241.997282_config.yaml:37]: Same as above

File: crs_process_data/data/surrogate1/toolpaths/swri_part/crs.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    force_tolerance: 7.5
+    force_controlled_trajectories: true
+    ur_tool_change_script: simpleMove.urp
+    execute_tool_change: false
+  part_registration:
+    simulation_pose: [0.0, 0.8, 0.2, -1.57, 3.14, -1.57]
+    seed_pose: [0.05, 0.95, 0.0, -1.57, 3.14, -1.57]
+    target_frame_id: world
+    part_file: /home/tmarr/crs_data/swri_part/swri_part.stl
+    toolpath_file: /home/tmarr/crs_data/swri_part/scan.yaml
```
Comment (jrgnicho) [inline crs_process_data/data/surrogate1/toolpaths/swri_part/crs.yaml:42]: remove the absolute path to the file

File: crs_support/urdf/crs.urdf.xacro [hunk 7]
```diff
@@ -433,45 +419,19 @@
...
+
+    <joint name="${prefix}sander_intermediate_to_sander_center_link" type="fixed">
+      <parent link="${prefix}sander_intermediate_link" />
+      <child link="${prefix}sander_center_link" />
+      <origin  xyz="0.0 0.0 0.0125" rpy="0.0 0.0 0.0"/>
+    </joint>
+
+    <joint name="${prefix}tool0_to_sander_intermediate_link" type="fixed">
+        <parent link="${prefix}eoat_link" />
+        <child link="${prefix}sander_intermediate_link" />
```
Comment (jrgnicho) [inline crs_support/urdf/crs.urdf.xacro:432]: Is this the configuration for the mockup end effector?

File: crs_support/urdf/swri_demo.urdf.xacro [hunk 8]
```diff
@@ -72,7 +72,7 @@
         <geometry>
           <box size="$(arg table_size) 0.1 $(arg table_size)"/>
         </geometry>
-      </collision>
+      </collision>-->
```
Comment (jrgnicho) [inline crs_support/urdf/swri_demo.urdf.xacro:75]: Should the walls be removed?

**Review by jrgnicho — CHANGES_REQUESTED** [review]

File: crs_support/urdf/crs.urdf.xacro (hunk 7 above)
Comment (marrts) [reply crs_support/urdf/crs.urdf.xacro:432]: I added this intermediate link to account for the sander pad which I am uncertain of it's size and this makes it easier to modify/account for that and I can easily add sander pad collision geometry. This is for the primary Spirit end effector, not the test sander bracket we are using at SwRI

File: crs_process_data/data/main/toolpaths/part_2/configs/job_0_degrees_20200520T192241.997282_config.yaml (hunk 4 above)
Comment (marrts) [reply crs_process_data/data/main/toolpaths/part_2/configs/job_0_degrees_20200520T192241.997282_config.yaml:37]: I think it automatically updates whenever you run it on your system

File: crs_process_data/data/surrogate1/toolpaths/swri_part/crs.yaml (hunk 6 above)
Comment (marrts) [reply crs_process_data/data/surrogate1/toolpaths/swri_part/crs.yaml:42]: Same as above

File: crs_motion_planning/src/path_planning_utils.cpp (hunk 3 above)
Comment (marrts) [reply crs_motion_planning/src/path_planning_utils.cpp:565]: it's supposed to mean both the retreat (ret) and approach (app) are reachable (addable). If either is not reachable then it tosses out that segment.

File: crs_motion_planning/src/motion_planning_server.cpp (hunk 1 above)
Comment (marrts) [reply crs_motion_planning/src/motion_planning_server.cpp:185]: Yes, that can be adjusted to be a parameter

File: crs_support/urdf/swri_demo.urdf.xacro (hunk 8 above)
Comment (marrts) [reply crs_support/urdf/swri_demo.urdf.xacro:75]: I removed that wall so I could test with that vertical aircraft part we have. The virtual walls were in that exact same spot

File: crs_process_data/data/main/toolpaths/part_2/configs/job_0_degrees_20200520T192241.997282_config.yaml (hunk 4 above)
Comment (jrgnicho) [reply crs_process_data/data/main/toolpaths/part_2/configs/job_0_degrees_20200520T192241.997282_config.yaml:37]: It shouldn't be doing that any longer, merged the change about a week or two ago

**Pull request closed without merging**